src/hathor.c

```diff
@@ -70,3 +91,65 @@ void hash160(unsigned char *in, unsigned short inlen, unsigned char *out) {
 void compress_public_key(unsigned char *value) {
     value[0] = ((value[64] & 1) ? 0x03 : 0x02);
 }
+
+int encode_base58(const unsigned char *in, size_t inlen, unsigned char *out, size_t outlen) {
+    unsigned char buffer[B58_MAX_INPUT_SIZE * 138 / 100 + 1] = {0};
+    size_t i = 0, j;
+    size_t startAt, stopAt;
+    size_t zeroCount = 0;
+    size_t outputSize;
+
+    if (inlen > B58_MAX_INPUT_SIZE) {
```

Comment:
Is it maximum size, or we could validate the exact size of the address?

Or is this just a generic method to encode base58 strings without validating addresses?

Reply:
This is a generic method. There's no address validation.